drivers/sdhc/brcm_bcm2712_sdhci.c

```diff
@@ -460,9 +465,36 @@ static int bcm2712_sdhci_init(const struct device *dev)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_BRCM_BCM2712_SDHC_USE_IRQ)) {
+		cfg->irq_config(dev);
+	}
+
 	return 0;
 }
 
+#if defined(CONFIG_BRCM_BCM2712_SDHC_USE_IRQ)
+static void bcm2712_irq_handler(const void *arg)
+{
+	const struct device *dev = arg;
+	struct bcm2712_sdhci_data *data = dev->data;
+
+	sdhc_irq_cb(&data->sdhci_ctx);
+}
+
+#define BCM2712_SDHC_IRQ_CFG(inst)                                                                 \
+	static void irq_config_##inst(const struct device *dev)                                    \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(inst), DT_INST_IRQ(inst, priority), bcm2712_irq_handler,  \
```

Comment:
According to the bindings - irq is not mandatory. This means that config can be enabled but no irqs were provided. In this case this code will fail. I think you should handle this situation.

Reply:
The Kconfig option makes this functionality optional and it will not even build in this case.

Reply:
But still it is possible from dts perspective to build with this parameter and do not provide any irq

Reply:
I truly do not understand what is the issue here :(
```
- CONFIG_BRCM_BCM2712_SDHC_USE_IRQ=n : interrupts=<ignore>. Build ok. Run ok.
- CONFIG_BRCM_BCM2712_SDHC_USE_IRQ=y : interrupts=<absent>. Build fail.
- CONFIG_BRCM_BCM2712_SDHC_USE_IRQ=y : interrupts=<present>. Build ok. Run ok.
```
Other drivers implemented the same way

Reply:
Other drivers can't be built without interrupts in dts

Reply:
I don't think that failed build is acceptable scenario. DT_INST_IRQ_HAS_IDX may be used

Reply:
With all do respect I'm not going to satisfy this request. If user want to use IRQ the correct DT configuration has to be provided and before selecting CONFIG_BRCM_BCM2712_SDHC_USE_IRQ. if configuration is incorrect build is failed which indicates an error. But propose to succeed the build and get binary which will have unexpected behavior.  The above code is absolutely normal practice for Zephyr and, if you wish you can check serial drivers and CONFIG_UART_INTERRUPT_DRIVEN as an example. Or rcar_mmc.c.

Reply:
Ok. I'm not going to complain. I disagree with your point "if configuration is incorrect build is failed " - when you don't provide irq configuration IS correct according to the bindings. So I see this as fail build with CORRECT configuration.
Leaving this to other reviewers.

Reply:
>  I disagree with your point "if configuration is incorrect build is failed " - when you don't provide irq configuration IS correct according to the bindings. 

Seems, You are mixing DT build vs full system build. DT build will always succeed while system depends on Kconfig settings, which consider IRQ support optional. And if IRQ is  enabled it's required for  IRQ to be provided also.

Reply:
If feature is enabled in Kconfig doesn't mean that I have to use it. Why can't I enable IRQ and use polling?

Reply:
After communication with @GrygiriiS we've decided to leave this implementation as is. So I have no complains about that